mitmproxy/addons/xss.py

```diff
@@ -0,0 +1,391 @@
+from mitmproxy import ctx  # Used for logging information to mitmproxy
+from socket import gaierror, gethostbyname  # Used to check whether a domain name resolves
+from urllib.parse import urlparse  # Used to modify paths and queries for URLs
+import requests  # Used to send additional requests when looking for XSSs
+import re  # Used for pulling out the payload
+from html.parser import HTMLParser  # used for parsing HTML
+
+# The actual payload is put between a frontWall and a backWall to make it easy
+# to locate the payload with regular expressions
+frontWall = b"1029zxc"
```

Comment:
We try to follow [PEP-8](https://www.python.org/dev/peps/pep-0008/) for the project where it makes sense.

For the whole file:
Variables and methods should be lowercase, with words separated by underscores as necessary to improve readability (e.g. `front_wall`). For constants you can also go for e.g. `FRONT_WALL`, if that does a better job at conveying your intention. Class names should use CapWords convention.